app/models/concerns/core_data_connector/nameable.rb

```diff
@@ -2,20 +2,35 @@ module CoreDataConnector
   module Nameable
     extend ActiveSupport::Concern
 
-    included do
-      has_many "#{self.model_name.param_key}_names".to_sym, dependent: :destroy
-      has_one :primary_name, -> { where(primary: true) }, class_name: "#{self.name}Name"
+    class_methods do
+      def name_table(name, options = nil)
+        if options && options[:polymorphic]
+          self.send(:has_many, name.to_sym, as: :nameable, dependent: :destroy)
+          has_one :primary_name, -> { where(primary: true) }, class_name: name.to_s.classify, as: :nameable
+        else
+          self.send(:has_many, name.to_sym, dependent: :destroy)
+          has_one :primary_name, -> { where(primary: true) }, class_name: name.to_s.classify
+        end
+
+        # Nested attributes
+        self.send(:accepts_nested_attributes_for, name.to_sym, allow_destroy: true)
 
-      # Nested attributes
-      accepts_nested_attributes_for "#{self.model_name.param_key}_names".to_sym, allow_destroy: true
+        @name_table = name
+      end
 
+      def get_name_table
```

Comment:
I spent some time trying to get `mattr_accessor` working (`cattr_accessor` has been removed and is now an alias for `mattr_accessor`) and I eventually got something that didn't error on me based on the examples from https://api.rubyonrails.org/v7.0.5/classes/Module.html#method-i-mattr_accessor.

However, I found two issues with this approach:

1. I had to call `model.class_variable_get("@@names_table")` every time I accessed it, which was more tedious than just `get_name_table`.
2. `mattr_accessor` only seemed to work if I set `names_table` as a class variable (`@@`), and it seems like setting a class variable from a concern sets it for all models that include the concern. In other words, People/Places functionality broke because the value was being set to `source_titles` by the source models.

If there's a way to get `mattr_accessor` to work on a properly scoped, per-model basis, I'm willing to try it, but I feel like I've burned too much time on a minor thing that might not actually come out simpler than `get_name_table` in the end.